src/PrestaShopBundle/Command/SecurityAnnotationLinterCommand.php

```diff
@@ -27,35 +27,155 @@
 namespace PrestaShopBundle\Command;
 
 use PrestaShopBundle\Routing\Linter\Exception\LinterException;
+use PrestaShopBundle\Routing\Linter\SecurityAnnotationLinter;
 use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\Style\SymfonyStyle;
 use Symfony\Component\Routing\Route;
 
 /**
  * Checks if all admin routes have @AdminSecurity configured
+ *
+ * @see \PrestaShopBundle\Security\Annotation\AdminSecurity
  */
 final class SecurityAnnotationLinterCommand extends ContainerAwareCommand
 {
+    const ACTION_LIST_ALL = 'list';
+    const ACTION_FIND_MISSING = 'find-missing';
+
+    /**
+     * @param string $expression
+     *
+     * @return string
+     */
+    public static function parseExpression($expression)
+    {
+        $pattern1 = '#\[(.*)\]#';
+        $pattern2 = '#is_granted\((.*),#';
+        $matches1 = [];
+        $matches2 = [];
+        preg_match($pattern1, $expression, $matches1);
+        preg_match($pattern2, $expression, $matches2);
+
+        if (count($matches1) > 1) {
+            return $matches1[1];
+        }
+        if (count($matches2) > 1) {
+            return $matches2[1];
+        }
```

Comment:
```suggestion

        if (count($matches1) > 1) {
            return $matches1[1];
        }
        preg_match($pattern2, $expression, $matches2);
        if (count($matches2) > 1) {
            return $matches2[1];
        }
```

Reply:
Still a fan of micro-optimization right 😁?

Reply:
If we can avoid a regex, it's not an optimization.

Reply:
It's a linter command that is currently being run once per release (= once or twice every 6 months) 😄

Reply:
Optimized code is good code, another code is code to optimize.

Reply:
We've had this discussion before 😛. If you want to say beautiful sentences, I will do it too:
- "Good is better than perfect"
- "Make it work, make it right, make it fast"

Reply:
Octogone?

Reply:
😄no I will do the requested changes because I want this approval 😛

Reply:
Not funny :(

Reply:
I know you want to see BLOOD ALL OVER THE PLACE @PierreRambaud but not this time 😄

Reply:
![https://media.giphy.com/media/xTiTntMlp6OVDR7j9e/source.gif](https://media.giphy.com/media/xTiTntMlp6OVDR7j9e/source.gif)